### ISSUE TITLE
Use Bitcoin types everywhere

### ISSUE THREAD

**Issue by KnowWhoami** [citadel-tech/coinswap#262]
* While updating our `rust-bitcoin` dependency, we attempted to replace all primitive types with the alternative `rust-bitcoin` types in https://github.com/citadel-tech/coinswap/pull/182.
But there are many places still left to apply these changes , thus created subtasks for them.

- [ ] Use `Amount` type for denoting Bitcoin amounts instead of plain `u64`.

  - Example: https://github.com/citadel-tech/coinswap/blob/57bf23866d01b876a2bab14b54ff3a01db385f1b/src/protocol/contract.rs#L57

- [ ] Use `FeeRate` type for fee-rate.

- [ ] Use `locktime::relative::LockTime` for denoting relative locktime instead of `u16`.

- [ ] Use `BlockHeight` for denoting the height of a block.

**Comment by A-y-ush** [comment]
Hey @KnowWhoami I would like to work on this issue. Can I be assigned to it ?

**Comment by Shash-Projects** [comment]
@KnowWhoami , Do we need to make these type changes across all the files ?

**Comment by mojoX911** [comment]
Yes this should be applied across all modules in the lib wherever applicable.

**Comment by Shash-Projects** [comment]
Can you please tell me whether the fee_rate is in satoshi per virtual byte (sat_per_vb) or per kilo virtual byte (sat_per_kvb) ?

**Comment by mojoX911** [comment]
`sat_per_vb` would be preferred.

**Comment by KnowWhoami** [comment]
> Can you please tell me whether the fee_rate is in satoshi per virtual byte (sat_per_vb) or per kilo virtual byte (sat_per_kvb) ?

> `sats_per_vb` would be preferred.

No, `Rust-bitcoin` have a dedicated struct [FeeRate](https://docs.rs/bitcoin/0.31.0/bitcoin/blockdata/fee_rate/struct.FeeRate.html)  for representing fee rate in `sat/(kilo weight unit)`.
Just use that struct instead.

**Comment by Shash-Projects** [comment]
Yes, but the FreeRate struct offers two methods: sat per vbyte and other sat per kilo vbyte. Should i go with per kilo vbyte?

**Comment by mojoX911** [comment]
Any one of them works. I would go for sat per vbyte as thats the defacto unit used in most places. And we will use that or our apps too.

**Comment by Shash-Projects** [comment]
Can you please review the pr#342 and notify the changes that need to be made ?

**Comment by KnowWhoami** [comment]
looking into #342

**Comment by Shash-Projects** [comment]
I think i can start working on this issue again. Should i open independent pr for updating types of amount, feeRate, locktime or just a single pr after making all the changes?

**Comment by KnowWhoami** [comment]
Single pr works.

**Comment by xTwo56** [comment]
may i give a shot?

**Comment by KnowWhoami** [comment]
yeah , go ahead..

**Comment by xTwo56** [comment]
In a lot of places, argument `fee_rate` is used with type `Amount` which is accepting an approx fee, 
should I rename arg `fee_rate` => `fee` with type Amount
e.g. 

```
pub(crate) fn create_receivers_contract_tx(
    input: OutPoint,
    input_value: Amount,
    contract_redeemscript: &ScriptBuf,
    fee_rate: FeeRate,
) -> Result<Transaction, ProtocolError> {

```
```
crate::protocol::contract::create_receivers_contract_tx(
    previous_funding_output,
    maker_funding_tx_value,
    next_contract_redeemscript,
    MINER_FEE, // <= Amount
  )},
```

if type `FeeRate` will be used, explicit conversion to Amount is still required which is just ugly to look at
```
        let calc_funding_tx_fees = FeeRate::from_sat_per_vb_unchecked(
            message.contract_feerate.to_sat_per_vb_ceil()
                * (message.next_coinswap_info.len() as u64),
        );

        let outgoing_amount = if let Some(a) = incoming_amount.checked_sub(Amount::from_sat(
            calc_coinswap_fees.to_sat() + calc_funding_tx_fees.to_sat_per_vb_ceil(),
        )) 
```

**Comment by xTwo56** [comment]
@mojoX911, @KnowWhoami, can someone review #454.
